### PR TITLE
fix: inherit AWS_DEFAULT_REGION into AWS_REGION

### PR DESCRIPTION
**Pull request by sublimino**
If AWS_REGION is not set then the AWS golang client doesn't function correctly.

This PR inherits that variable from AWS_DEFAULT_REGION if it is set.

It also refactors launch-entrypoint to:
- run in Bash safe mode
- exec bash instead of calling it
- add a `main()` block at the top of the file for readability
- write an initial alias (`ll` as `ls -lasp`)
- configure .inputrc for single-tab completion

### CODE REVIEW

**Comment by jondkent** [comment]
This looks visually fine to me, I'll leave to @raoulmillais to review more completely

**Comment by raoulmillais** [comment]
This is great, thanks @sublimino :) we ought to update the docs in [./doc-templates/README.md] to reflect that we have made every effort to solve this snag. I also wonder if we can test this from tcl in the acceptance tests? (we haven't tested these scripts so far)

While we're nitpicking, those variable names in the `box` function make me cry!

**Comment by sublimino** [comment]
Let me get the docs and finish the vars 🙏

**Comment by raoulmillais** [comment]
Sorry that doc link didnt work ... it's ./doc-templates/README.template.md - you should commit and then run `make docs` and commit again to separate the  doc generation from the actual changes

**Comment by raoulmillais** [comment]
See #51 for documenting the documentation

**Comment by sublimino** [comment]
Updated docs and tests, but I can't run `make docs` -- pseudo-version errors locally. Can somebody do that for me please?

I also don't have a passing test suite on my machine, but didn't verify that before the PR! If anybody can I'd be grateful, otherwise I'll get back in and test from scratch :pray:

**Comment by raoulmillais** [comment]
`make docker-test` shows your new tests failing for me - but this is because `launch-entrypoint.sh` generates the files and goss doesnt actually tun the entrypoint, `make docker-test` overrides the entrypoint and runs `acceptance.sh` which does a `goss validate`

File: goss.yaml [hunk 1]
```diff
@@ -26,6 +26,9 @@ command:
     exec: "which {{.}}"
     exit-status: 0
 {{end}}
+  "AWS_PROFILE environment variable is not empty":
```
Comment (raoulmillais) [inline goss.yaml]: This should be AWS_REGION

**Comment by raoulmillais** [comment]
> Updated docs and tests, but I can't run `make docs` -- pseudo-version errors locally. Can somebody do that for me please?
> 
> I also don't have a passing test suite on my machine, but didn't verify that before the PR! If anybody can I'd be grateful, otherwise I'll get back in and test from scratch pray

We need to remove godocdown... It is unmaintained and has recently broken: it doesn't use go modules and a downstream dependcy is no longer available.  I'll submit an issue to remove the API doc generation and we'll use godoc for API docs

File: launch-entrypoint.sh [hunk 2]
```diff
@@ -1,25 +1,61 @@
...
+}
+
+fix_aws_environment_variables() {
+  export AWS_REGION="${AWS_REGION:-${AWS_DEFAULT_REGION:-}}"
+}
+
+write_bash_aliases() {
+  cat >>/home/launch/.bash_aliases <<EOF
+alias ll='ls -lasp'
+EOF
```
Comment (raoulmillais) [inline launch-entrypoint.sh]: I think we should just include a static .bash_aliases file rather than write it every time we run the container

Comment (raoulmillais) [reply launch-entrypoint.sh]: Same for inputrc

Comment (raoulmillais) [reply launch-entrypoint.sh]: Also the `>>` means the aliases will grow duplicates on each run

**Comment by raoulmillais** [comment]
> > Updated docs and tests, but I can't run `make docs` -- pseudo-version errors locally. Can somebody do that for me please?
> > I also don't have a passing test suite on my machine, but didn't verify that before the PR! If anybody can I'd be grateful, otherwise I'll get back in and test from scratch pray
> 
> We need to remove godocdown... It is unmaintained and has recently broken: it doesn't use go modules and a downstream dependency is no longer available. I'll submit an issue to remove the API doc generation and we'll use godoc for API docs

I've updated #51 to include removing `godocdown`

**Comment by raoulmillais** [comment]
This is great thanks @sublimino

**Pull request merged**